doc/Manuals/LoadBalancing/ig-xlb_x-road_external_load_balancer_installation_guide.md

```diff
@@ -0,0 +1,830 @@
+periodically (once per minute) and before the services are started. That means that during boot, if the master server is
+available, the configuration will be synchronized before the `xroad-proxy` service is started. If the master node is down,
+there will be a small delay before the services are started.
+
+> Note that only modifications to the signer keyconf will be applied when the system is running. Changes to any other
+configuration files,  like `local.ini`, require restarting the services, which is not automatic.
+
+#### 5.2.1 RHEL: Use systemd for configuration synchronization
+
+First, add `xroad-sync` as a `systemd` service.
+
+Create a new file `/etc/systemd/system/xroad-sync.service`:
+
+```
+[Unit]
+Description=X-Road Sync Task
+After=network.target
+Before=xroad-proxy.service
+Before=xroad-signer.service
+Before=xroad-confclient.service
+Before=xroad-jetty.service
+[Service]
+User=xroad
+Group=xroad
+Type=oneshot
+Environment=XROAD_USER=xroad-slave
+Environment=MASTER=<master_host>
+ExecStart=/usr/bin/rsync -e "ssh -o ConnectTimeout=5 " -aqz --timeout=10 --delete-delay --exclude db.properties --exclude "/conf.d/node.ini" --exclude "*.tmp" --exclude "/postgresql" --exclude "/nginx" --exclude "/globalconf" --exclude "/jetty" --delay-updates --log-file=/var/log/xroad/slave-sync.log ${XROAD_USER}@${MASTER}:/etc/xroad/ /etc/xroad/
+[Install]
+WantedBy=multi-user.target
+WantedBy=xroad-proxy.service
+```
+Where `<master_host>` is the DNS name or IP address of the master node.
+
+The service will log `rsync` events to `/var/log/xroad/slave-sync.log`.
+
+Then, add a timer for periodic updates.
+
+Create a new file `/etc/systemd/system/xroad-sync.timer`:
+
+```
+[Unit]
+Description=Sync X-Road configuration
+[Timer]
+OnBootSec=60
+OnUnitActiveSec=60
+[Install]
+WantedBy=timers.target
+```
+
+Next, configure SELinux to allow `rsync` to be run as a `systemd` service
+
+```
+setsebool -P rsync_client 1
+setsebool -P rsync_full_access 1
+```
+Finally, enable the services:
+```
+systemctl enable xroad-sync.timer xroad-sync.service
+systemctl start xroad-sync.timer
+```
+
+#### 5.2.2 Ubuntu: Use upstart and cron for configuration synchronization
+
+First, create the main upstart task for syncing.
+
+Create a new file: `/etc/init/xroad-sync.conf`:
+```
+# xroad-sync
+env XROAD_USER=xroad-slave
+env MASTER=<master_host>
+task
+script
+  su -s /bin/sh -c 'exec "$0" "$@"' xroad -- rsync -e "ssh -o ConnectTimeout=5 " -aqz --timeout=10 --delete-delay --exclude db.properties --exclude "/conf.d/node.ini" --exclude "*.tmp" --exclude "/postgresql" --exclude "/nginx" --exclude "/globalconf" --exclude "/jetty" --delay-updates --log-file=/var/log/xroad/slave-sync.log ${XROAD_USER}@${MASTER}:/etc/xroad/ /etc/xroad/
+end script
+```
+Where `<master_host>` is the DNS name or IP address of the master node.
+
+The task will log `rsync` events to `/var/log/xroad/slave-sync.log`.
```

Comment:
I get the following errors:
2017/04/12 16:04:01 [16296] rsync: connection unexpectedly closed (0 bytes received so far) [Receiver]
2017/04/12 16:04:01 [16296] rsync error: error in rsync protocol data stream (code 12) at io.c(226) [Receiver=3.1.0]

Can it be that some steps are missing? ssh xroad-slave@<master_host> works fine (but of course disconnects right after connection because xroad-slave has /bin/false shell)